src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -337,6 +470,60 @@ private void SendMessageAndListenAndReportTestCases(IEnumerable<string> sources,
             this.testPlatformEventSource.TranslationLayerDiscoveryStop();
         }
 
+        private async Task SendMessageAndListenAndReportTestCasesAsync(IEnumerable<string> sources, string runSettings, ITestDiscoveryEventsHandler eventHandler)
+        {
+            try
+            {
+                this.communicationManager.SendMessage(
+                            MessageType.StartDiscovery,
+                            new DiscoveryRequestPayload() { Sources = sources, RunSettings = runSettings },
+                            this.protocolVersion);
+                var isDiscoveryComplete = false;
+
+                // Cycle through the messages that the vstest.console sends.
+                // Currently each of the operations are not separate tasks since they should not each take much time.
+                // This is just a notification.
+                while (!isDiscoveryComplete)
```

Comment:
Since the calls will not be blocking with the event based model, we need to consider how they will play with the async interface.